libshpool/src/daemon/shell.rs

```diff
@@ -213,26 +207,19 @@ impl SessionInner {
         let daily_messenger = Arc::clone(&self.daily_messenger);
         let mut needs_initial_motd_dump = self.needs_initial_motd_dump;
 
-        let vterm_width = {
-            let config = self.config.get();
-            config.vt100_output_spool_width.unwrap_or(VTERM_WIDTH)
-        };
         let mut pty_master = self.pty_master.is_parent()?;
         let watchable_master = pty_master;
         let name = self.name.clone();
-        let mut closure = move || {
+        let config = self.config.clone();
+        let closure = move || {
             let _s = span!(Level::INFO, "shell->client", s = name, cid = args.conn_id).entered();
 
-            let mut output_spool =
-                if matches!(args.session_restore_mode, config::SessionRestoreMode::Simple) {
-                    None
-                } else {
-                    Some(shpool_vt100::Parser::new(
-                        args.tty_size.rows,
-                        vterm_width,
-                        args.scrollback_lines,
-                    ))
-                };
+            let mut output_spool = session_restore::new(
+                config,
+                &args.session_restore_mode,
```

Comment:
I'm realizing that we should probably be dynamically fetching this from the config rather than passing down an explicit value since this is set in the config file (https://github.com/shell-pool/shpool/blob/d5e62545c37aefd4ecf7c768e57908c81332203b/libshpool/src/config.rs#L241). No need to fix it in this patch since it is an existing issue, but we should probably at least leave a TODO.